ros_packages/voice_assistant/voice_assistant/chat.py

```diff
@@ -246,13 +294,53 @@ async def chat(self, goal_handle: ServerGoalHandle):
             goal_handle.abort()
             return Chat.Result()
 
+        # for storing and analyzing data from the public-api
+        curr_sentence: str = ""
+        prev_sentence: str | None = None
+        sentence_boundary = re.compile(r"[^\d | ^A-Z][\.|!|\?|:]")
+
+        # process incoming data from public-api
+        bool_update_chat_message: bool = False
+        for token in tokens:
+            # if the goal was cancelled, return immediately
+            if goal_handle.is_cancel_requested:
+                goal_handle.canceled()
+                return Chat.Result(rest=curr_sentence)
+            # if a sentence was already found and another token was received, forward the sentence as feedback
+            if prev_sentence is not None:
+                self.executor.create_task(
+                    self.create_chat_message,
+                    chat_id,
+                    prev_sentence,
+                    False,
+                    bool_update_chat_message,
+                    not bool_update_chat_message,
+                )
+                bool_update_chat_message = True
+                feedback = Chat.Feedback()
+                feedback.sentence = prev_sentence
+                goal_handle.publish_feedback(feedback)
+                prev_sentence = None
+            # check if the current token marks the end of a sentence
+            if sentence_boundary.search(curr_sentence):
+                prev_sentence = curr_sentence.strip()
+                curr_sentence = ""
+            curr_sentence += token
+
         # create chat-message for remaining input
-        if len(curr_text) > 0:
+        if len(curr_sentence) > 0:
             self.executor.create_task(
-                self.create_chat_message, chat_id, curr_text, False
+                self.create_chat_message,
+                chat_id,
+                curr_sentence,
+                False,
+                bool_update_chat_message,
+                bool_update_chat_message,
             )
-
-        # return the rest of the received text, that has not been forwarded as feedback
+        # return the restult
+        bool_update_chat_message = False
```

Comment:
Since the variable isn't used from this point on anymore and is scoped to the method, this statement can be removed.

Reply:
Removed